16.0 mig l10n fr ecotaxe

mourad-ehm opened the pull request.

mourad-ehm committed:
Add custome profile module + account config

mourad-ehm committed:
FIX bug compute ecotaxe

mourad-ehm committed:
Add ecotaxe classification

mourad-ehm committed:
ADD eco taxes classification

mourad-ehm committed:
Remove commented code

mourad-ehm committed:
[FIX] bug non compliant picking location dest

mourad-ehm committed:
Add base exotaxe (weight or qty) + change relation many2many between product and ecotaxe classification

mourad-ehm committed:
[FIX] compute generic base for account.invoice.taxe

mourad-ehm committed:
[FIX] bug off binary fileds + code format

mourad-ehm committed:
[FIX] move ecotaxte to product template

mourad-ehm committed:
[FIX] Clean Code

mourad-ehm committed:
[FIX] Add on change ecotaxe classification on product template

mourad-ehm committed:
[FIX] pylint code

mourad-ehm committed:
Fix Typo

Commit:
typo

mourad-ehm committed:
[FIX] Clean code

mourad-ehm committed:
[FIX] add README.rst

bealdav committed:
[IMP] add decimal precision to ecotaxe_coef

mourad-ehm committed:
[FIX] bugs and remarks from code review

oca-transbot committed:
OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

sbidoul committed:
remove obsolete .pot files [ci skip]

oca-travis committed:
[UPD] Update l10n_fr_ecotaxe.pot

StefanRijnhart committed:
[FIX] Pylint

mourad-ehm committed:
[MIG] module l10n_fr_ecotaxe to 12.0

mourad-ehm committed:
[FIX] split l10n_fr_ecotaxe in 3 modules (add l10n_fr_ecotaxe_sale, l10n_fr_ecotaxe_purchase)

mourad-ehm committed:
[FIX] add test

mourad-ehm committed:
[FIX] add sale & purchase ecotaxe test + format code

mourad-ehm committed:
[FIX] readme files

mourad-ehm committed:
[IMP] product view

mourad-ehm committed:
[FIX] add model _description

SilvioC2C committed:
[MIG] l10n_fr_ecotaxe: Migration to 15.0

SilvioC2C committed:
[IMP] l10n_fr_ecotaxe: black, isort, prettier

Commit:
[UPD] Update l10n_fr_ecotaxe.pot

OCA-git-bot committed:
[UPD] README.rst

ajaniszewska-dev committed:
Translated using Weblate (French)

Currently translated at 71.6% (38 of 53 strings)

Translation: l10n-france-15.0/l10n-france-15.0-l10n_fr_ecotaxe
Translate-URL: https://translation.odoo-community.org/projects/l10n-france-15-0/l10n-france-15-0-l10n_fr_ecotaxe/fr/

gurneyalex committed:
fixup! [MIG] l10n_fr_ecotaxe_sale: Migration to 15.0

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 47.1% (25 of 53 strings)

Translation: l10n-france-15.0/l10n-france-15.0-l10n_fr_ecotaxe
Translate-URL: https://translation.odoo-community.org/projects/l10n-france-15-0/l10n-france-15-0-l10n_fr_ecotaxe/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 47.1% (25 of 53 strings)

Translation: l10n-france-15.0/l10n-france-15.0-l10n_fr_ecotaxe
Translate-URL: https://translation.odoo-community.org/projects/l10n-france-15-0/l10n-france-15-0-l10n_fr_ecotaxe/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 47.1% (25 of 53 strings)

Translation: l10n-france-15.0/l10n-france-15.0-l10n_fr_ecotaxe
Translate-URL: https://translation.odoo-community.org/projects/l10n-france-15-0/l10n-france-15-0-l10n_fr_ecotaxe/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 47.1% (25 of 53 strings)

Translation: l10n-france-15.0/l10n-france-15.0-l10n_fr_ecotaxe
Translate-URL: https://translation.odoo-community.org/projects/l10n-france-15-0/l10n-france-15-0-l10n_fr_ecotaxe/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 47.1% (25 of 53 strings)

Translation: l10n-france-15.0/l10n-france-15.0-l10n_fr_ecotaxe
Translate-URL: https://translation.odoo-community.org/projects/l10n-france-15-0/l10n-france-15-0-l10n_fr_ecotaxe/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 47.1% (25 of 53 strings)

Translation: l10n-france-15.0/l10n-france-15.0-l10n_fr_ecotaxe
Translate-URL: https://translation.odoo-community.org/projects/l10n-france-15-0/l10n-france-15-0-l10n_fr_ecotaxe/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 47.1% (25 of 53 strings)

Translation: l10n-france-15.0/l10n-france-15.0-l10n_fr_ecotaxe
Translate-URL: https://translation.odoo-community.org/projects/l10n-france-15-0/l10n-france-15-0-l10n_fr_ecotaxe/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 47.1% (25 of 53 strings)

Translation: l10n-france-15.0/l10n-france-15.0-l10n_fr_ecotaxe
Translate-URL: https://translation.odoo-community.org/projects/l10n-france-15-0/l10n-france-15-0-l10n_fr_ecotaxe/it/

mourad-ehm committed:
[IMP] l10n_fr_ecotaxe: pre-commit execution

mourad-ehm committed:
[MIG] l10n_fr_ecotaxe: Migration to 16.0

mourad-ehm committed:
[FIX][16.0] l10n_fr_ecotaxe: fix views &  test

[14.0][IMP] l10n_fr_ecotaxe: manage multi ecotaxe classification by product

[14.0][FIX] l10n_fr_ecotaxe: pre-commit, access rule, view

[14.0][FIX] l10n_fr_ecotaxe: imp multi ecotaxe classification on product form & edit ecotaxe line on account move line

[14.0][FIX] l10n_fr_ecotaxe: add archived web_ribbon on ecotaxe_classification & ecotaxe_category

[14.0][FIX] l10n_fr_ecotaxe: fix tests

[14.0][FIX] l10n_fr_ecotaxe: fix _onchange_product_ecotaxe_line

[16.0][FIX] fix pre-commit & typo

[16.0][FIX] update readme files

legalsylvain commented:
/ocabot migration l10n_fr_ecotaxe

pfranck commented:
Any help needed to merge the request ?

pfranck commented:
Any solution to be able to merge this request

florian-dacosta commented on l10n_fr_ecotaxe/models/account_ecotaxe_classification.py:
```diff
@@ -0,0 +1,78 @@
+    ecotaxe_scale_code = fields.Char()
+
+    @api.onchange("ecotaxe_type")
+    def _onchange_ecotaxe_type(self):
```
Would be nice to convert `default_fixed_ecotaxe` and `ecotaxe_coef` as readonly=False computed fields to replace this on change.

florian-dacosta commented on l10n_fr_ecotaxe/models/ecotaxe_line_product.py:
```diff
@@ -0,0 +1,51 @@
+
+
+class EcotaxeLineProduct(models.Model):
+    """class for objects which can be used to save mutili ecotaxe calssification  by product."""
```
mutili = multi

florian-dacosta commented on l10n_fr_ecotaxe/models/ecotaxe_line_product.py:
```diff
@@ -0,0 +1,51 @@
+        string="Ecotaxe Classification",
+    )
+    force_ecotaxe_amount = fields.Monetary(
+        help="Force ecotaxe.\n" "Allow to subtite default Ecotaxe Classification\n"
```
subtite = substitute ?

florian-dacosta reviewed (commented):
I did a quick code review, just some minor remarks.
When the PR is ready I will do some tests

mourad-ehm committed:
[16.0][IMP] manage ecotaxe for product variants

mourad-ehm committed:
[16.0][IMP] add ecotaxe sector and collector

bealdav commented on l10n_fr_ecotaxe/views/product_view.xml:
```diff
@@ -0,0 +1,55 @@
+                            readonly="1"
+                        >
+                            <tree editable="bottom">
+                                <field name="ecotaxe_classification_id" />
```
Just for usability

We already know it's Ecotax because this is the name of th o2m
```suggestion
                                <field name="ecotaxe_classification_id" string="Classification" />
```

bealdav commented on l10n_fr_ecotaxe/views/product_view.xml:
```diff
@@ -0,0 +1,55 @@
+                            <tree editable="bottom">
+                                <field name="ecotaxe_classification_id" />
+                                <field name="force_ecotaxe_amount" />
+                                <field name="ecotaxe_amount" />
```
```suggestion
                                <field name="ecotaxe_amount" string="Amount" />
```

bealdav commented on l10n_fr_ecotaxe/views/product_view.xml:
```diff
@@ -0,0 +1,55 @@
+                            <tree editable="bottom">
+                                <field name="ecotaxe_classification_id" />
+                                <field name="force_ecotaxe_amount" />
+                                <field name="ecotaxe_amount" />
```
Same here for name field

bealdav reviewed (commented):
Partial review

bealdav commented on l10n_fr_ecotaxe/models/ecotaxe_line_mixin.py:
```diff
@@ -0,0 +1,58 @@
+
+    product_id = fields.Many2one("product.product", string="Product", readonly=True)
+    currency_id = fields.Many2one("res.currency", string="Currency")
+    ecotaxe_classification_id = fields.Many2one(
```
Because it's an ecotaxe model, maybe `classification_id` name is sufficient here

bealdav commented on l10n_fr_ecotaxe/models/ecotaxe_line_mixin.py:
```diff
@@ -0,0 +1,58 @@
+        "account.ecotaxe.classification",
+        string="Ecotaxe Classification",
+    )
+    ecotaxe_amount_unit = fields.Monetary(
```
Because it's an ecotaxe model, maybe `amount_unit` name is sufficient here.

The same for ecotaxe name in fields below

florian-dacosta commented on l10n_fr_ecotaxe/models/account_move_line_ecotaxe.py:
```diff
@@ -0,0 +1,26 @@
+
+class AccountMoveLineEcotaxe(models.Model):
+    _name = "account.move.line.ecotaxe"
+    _inherit = "ecotaxe.line.mixin"
```
Missing description

mourad-ehm committed:
[16.0][FIX] l10n_fr_ecotaxe: don't display ecotaxe on vendor invoices

florian-dacosta committed:
[FIX] Add read rights for reaonly accounting group

mourad-ehm committed:
[16.0][FIX] pre-commit

[16.0][FIX] product variant view

[16.0][FIX] add unique sql constaint for ecotaxe classif on product

mourad-ehm committed:
[16.0][IMP] rename fields in l10n_fr_ecotaxe module

[16.0][IMP]  rename fields in l10n_fr_ecotaxe module rest of modif & imp product view

metaminux commented on l10n_fr_ecotaxe/models/product_product.py:
```diff
@@ -0,0 +1,75 @@
+        "ecotaxe.line.product",
+        compute="_compute_all_ecotaxe_line_product_ids",
+        search="_search_all_ecotaxe_line_product_ids",
+        string="Additional ecotaxe lines",
```
```suggestion
        string="All ecotaxe lines",
```
To avoid Warning : `Two fields (all_ecotaxe_line_product_ids, additional_ecotaxe_line_product_ids) of product.product() have the same label: Additional ecotaxe lines.`

mourad-ehm committed:
[16.0][FIX] fix amount digits precision & view update error

[16.0][FIX] fix psot-migration view load bug

[16.0][FIX] fix digits precision for ecotaxe amount on invoice & product & fixed ecotaxe

[16.0][FIX] tests

[16.0][FIX] replace onchange by compute

[16.0][FIX] pre-commit

bealdav reviewed (approved):
lgtm

metaminux reviewed (approved):
LGTM too

gurneyalex commented:
/ocabot merge nobump

bealdav commented:
Oops these modules has been hardly modified here

https://github.com/OCA/account-fiscal-rule/pull/428
https://github.com/OCA/account-fiscal-rule/pull/429

Inside Ecotax use tax odoo mechanism

gurneyalex commented:
hello @bealdav sorry I had missed the discussion on https://github.com/OCA/l10n-france/issues/483 and the PRs you mention. 

I just updated https://github.com/OCA/l10n-france/issues/373 and https://github.com/OCA/l10n-france/issues/487 with the information of the moving the modules. 

I'm not sure what the best course is here.

gurneyalex commented:
https://github.com/OCA/l10n-france/pull/552 is reverting this

bealdav commented:
It's our own fault. We had not a really good communication on it, sorry.
